source/include/azure_iot_config_defaults.h

```diff
@@ -0,0 +1,124 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+#ifndef AZURE_IOT_CONFIG_DEFAULTS_H
+#define AZURE_IOT_CONFIG_DEFAULTS_H
+
+#ifndef azureiotDEFAULT_TOKEN_TIMEOUT_IN_SEC
+    #define azureiotDEFAULT_TOKEN_TIMEOUT_IN_SEC        ( 60 * 60U )
+#endif
+
+#ifndef azureiotKEEP_ALIVE_TIMEOUT_SECONDS
+    #define azureiotKEEP_ALIVE_TIMEOUT_SECONDS          ( 60U )
+#endif
+
+#ifndef azureiotCONNACK_RECV_TIMEOUT_MS
+    #define azureiotCONNACK_RECV_TIMEOUT_MS             ( 1000U )
+#endif
+
+#ifndef azureiotSUBACK_WAIT_INTERVAL_MS
+    #define azureiotSUBACK_WAIT_INTERVAL_MS             ( 10U )
+#endif
+
+#ifndef azureiotUSERNAME_MAX
+    #define azureiotUSERNAME_MAX                        ( 128U )
+#endif
+
+#ifndef azureiotPASSWORD_MAX
+    #define azureiotPASSWORD_MAX                        ( 256U )
+#endif
+
+#ifndef azureiotTOPIC_MAX
+    #define azureiotTOPIC_MAX                           ( 128U )
+#endif
+
+#ifndef azureiotPROVISIONING_RESPONSE_MAX
+    #define azureiotPROVISIONING_RESPONSE_MAX           ( 512U )
+#endif
+
+#ifndef azureiotPROVISIONING_REQUEST_PAYLOAD_MAX
+    #define azureiotPROVISIONING_REQUEST_PAYLOAD_MAX    ( 512U )
+#endif
+
+/**
+ * @brief Milliseconds per second.
+ */
+#define azureiotMILLISECONDS_PER_SECOND                 ( 1000U )
+
+/**
+ * @brief Macro that is called in the AzureIoT Middleware library for logging "Error" level
+ * messages.
+ *
+ * To enable error level logging in the AzureIoT Middleware library, this macro should be mapped to the
```

Comment:
It's a bit more nuanced, because I can also enable enable this by `#define LogError` where I'm not necessarily the one mapping (which I'd think as my app code doing the `#define AZLogError <whatever>`).

Maybe something like below, with your word-smithing love of course and making sure my understanding of the FreeRTOS log setup is right.  I don't think our docs should explicitly say here #define LIBRARY_LOG_LEVEL or above, because that might change (albeit unlikely) but we can at least get folks in right direction.
```
To enable error level logging in the AzureIoT Middleware library either:
* Map the macro to the application-specific logging implementation that supports error logging.
* Enable FreeRTOS's logging stack by `#include <logging_stack.h>` and configuring its logging level to be at least error level.
``

Similar feedback for other log levels.